nearcore/src/state_sync.rs

```diff
@@ -264,7 +314,74 @@ async fn state_sync_dump(
                         None
                     }
                     Ok((state_root, num_parts, sync_prev_prev_hash)) => {
-                        match get_missing_part_ids_for_epoch(
+                        // Upload header
```

Comment:
This is really nested and hard to parse, can you refactor by returning early where possible and/or moving to helper methods?

Reply:
I will try my best to make it as clean as possible but there is a limitation on passing `chain` instance between async functions. 
[This](https://github.com/near/nearcore/commit/c42391481b5e82f0d8b780134e560f1f93978a8b) does not work because of `chain`.